lightnode/kvstore/storeutil/thread_safe_wrapper.go

```diff
@@ -0,0 +1,79 @@
+package storeutil
+
+import (
+	"github.com/Layr-Labs/eigenda/lightnode/kvstore"
+	"github.com/syndtr/goleveldb/leveldb/iterator"
+	"sync"
+)
+
+var _ kvstore.Store = &threadSafeStore{}
+
+// threadSafeStore is a wrapper around a KVStore that makes store access thread safe.
```

Comment:
The KV databases are usually thread safe already

Reply:
Most off the shelf implementations probably are, but it's possible to imagine one that isn't. For example, the simple in-memory implementation I wrote for unit testing is not thread safe. We probably don't have any production use cases for the thread safe wrapper in the short term, do you think I should delete it?

Reply:
If there is no anticipation of production use, it may be cleaner just make the in-memory KV store itself thread safe. Having this threadSafeStore feels like we are anticipating some prod KV DB to be thread unsafe.

Reply:
change made